core/example/benchmark/Cabana_CommPerformance.cpp

```diff
@@ -0,0 +1,573 @@
+                                          0.10,
+                                          0.5 };
+    int num_fraction = comm_fraction.size();
+
+    // Number of bytes we will send to each neighbor.
+    std::vector<std::size_t> comm_bytes( num_fraction );
+
+    // DISTRIBUTOR
+    // -----------
+
+    // Create distributor timers.
+    ParallelTimer distributor_fast_create(
+        comm, test_prefix+"distributor_fast_create", num_fraction );
+    ParallelTimer distributor_slow_create(
+        comm, test_prefix+"distributor_slow_create", num_fraction );
+    ParallelTimer distributor_aosoa_migrate(
+        comm, test_prefix+"distributor_aosoa_migrate", num_fraction );
+    ParallelTimer distributor_slice_migrate(
+        comm, test_prefix+"distributor_slice_migrate", num_fraction );
+
+    // Loop over comm fractions.
+    for ( int fraction = 0; fraction < num_fraction; ++fraction )
+    {
+        // Create the migrate distribution in the data memory space. This is
+        // where it would likely be created.
+        int num_send = num_particle * comm_fraction[fraction];
+        int send_per_neighbor = num_send / 26;
+        num_send = send_per_neighbor * 26;
+        int num_stay = num_particle - num_send;
+        comm_bytes[fraction] = send_per_neighbor * bytes_per_particle;
+        Kokkos::View<int*,Kokkos::HostSpace>
+            export_ranks_host( "export_ranks", num_particle );
+        for ( int p = 0; p < num_stay; ++p )
+            export_ranks_host(p) = neighbor_ranks[0];
+        for ( int n = 0; n < 26; ++n )
+            for ( int p = 0; p < send_per_neighbor; ++p )
+                export_ranks_host( num_stay + n * send_per_neighbor + p ) =
+                    neighbor_ranks[n+1];
+        auto export_ranks = Kokkos::create_mirror_view_and_copy(
+            data_memory_space(), export_ranks_host );
+
+        // Run tests and time the ensemble.
+        for ( int t = 0; t < num_run; ++t )
+        {
+            // Create source particles.
+            aosoa_type src_particles( num_particle );
+
+            // Create destination particles.
+            aosoa_type dst_particles;
+
+            // Create a distributor using the fast construction method.
+            distributor_fast_create.start(fraction);
+            auto comm_export_ranks =
+                Kokkos::create_mirror_view_and_copy(
+                    comm_memory_space(), export_ranks );
+            Cabana::Distributor<comm_memory_space> distributor_fast(
+                comm, comm_export_ranks, neighbor_ranks );
+            distributor_fast_create.stop(fraction);
+
+            // Create a distributor using the slow construction method.
+            distributor_slow_create.start(fraction);
+            comm_export_ranks =
+                Kokkos::create_mirror_view_and_copy(
+                    comm_memory_space(), export_ranks );
+            Cabana::Distributor<comm_memory_space> distributor_slow(
```

Comment:
In an ideal world this wouldn't be `slow`, right? More.. `simple` or something?

Reply:
no it's actually slower because you need to compute the communication topology rather than having received it from the user. maybe the better terminology is "general"

Reply:
changed "slow" -> "general"